GUI/Arduino/UMSATS_GUI.ino

```diff
@@ -244,67 +304,94 @@ void printFrame(CAN_FRAME &frame) {
    }
    Serial.print("\r\n");
 }
+
+
 //=========================================================================
 //LOOP
 //=========================================================================
+
 void loop(){
+
   CAN_FRAME incoming;
-  static unsigned long lastTime = 0;
-//  STATE = NOT_CONNECTED;
-//  Serial.println(CommsStateMachineInput);
-if(CommsStateMachineInput == 'a'){
-  radio();}
-//  while(Can0.available() <= 0){}
+
+  if(comms_sm_input == 'a'{
+    radio();
+  }
+
   if(Can0.available() > 0){
     
     Can0.read(incoming);
     
     printFrame(incoming);  
   }
+
   rx = Serial.parseInt();
-switch(rx){
-  
-  case POWER:
-    while(Serial.available()<= 0){}
-    value = Serial.parseInt();
-   
-    sendPower(value);
-    break;
-    case PAYLOAD:
-    
-    while(Serial.available()<= 0){}
-    well = Serial.parseInt();
-    while(Serial.available()<= 0){}
-    value = Serial.parseInt();
-    sendPayload(well,value);
-    break;
-    case GROUND:
-    while(Serial.available()<= 0){}
-    command = Serial.parseInt();
-    while(Serial.available()<= 0){}
-    well = Serial.parseInt();
-    while(Serial.available()<= 0){}
-    value = Serial.parseInt();
-    sendGround(command,well,value);
-    break;
-    case COMMS:
-    
-    while(Serial.available()<= 0){}
-    command = Serial.parseInt();
-    switch (command){
-          
-        case 0: CommsStateMachineInput = 'a';
-                Serial.println("Starting radio demo.");
-                break;
-        case 1: CommsStateMachineInput = 'b';
-                Serial.println("Stopping radio demo..");
-                break;
-    }
-    
-    break;
-    
 
-  
-  }
+	switch(rx){
```

Comment:
Looks like there is a mix of tabs / spaces that's causing the spacing to be weird. This is indented as a tab, whereas the line before is indented as spaces. You can do a "replace all" fairly easily on all the tabs if you open up this file in Notepad++ (i.e. replace '\t' with two or four spaces).